pkg/backup/item_backupper.go

```diff
@@ -33,62 +33,16 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	kubeerrs "k8s.io/apimachinery/pkg/util/errors"
 
-	api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
 	"github.com/vmware-tanzu/velero/pkg/client"
 	"github.com/vmware-tanzu/velero/pkg/discovery"
 	"github.com/vmware-tanzu/velero/pkg/kuberesource"
 	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
-	"github.com/vmware-tanzu/velero/pkg/podexec"
 	"github.com/vmware-tanzu/velero/pkg/restic"
 	"github.com/vmware-tanzu/velero/pkg/volume"
 )
 
-type itemBackupperFactory interface {
-	newItemBackupper(
-		backup *Request,
-		podCommandExecutor podexec.PodCommandExecutor,
-		tarWriter tarWriter,
-		dynamicFactory client.DynamicFactory,
-		discoveryHelper discovery.Helper,
-		resticBackupper restic.Backupper,
-		resticSnapshotTracker *pvcSnapshotTracker,
-		volumeSnapshotterGetter VolumeSnapshotterGetter,
-	) ItemBackupper
-}
-
-type defaultItemBackupperFactory struct{}
-
-func (f *defaultItemBackupperFactory) newItemBackupper(
-	backupRequest *Request,
-	podCommandExecutor podexec.PodCommandExecutor,
-	tarWriter tarWriter,
-	dynamicFactory client.DynamicFactory,
-	discoveryHelper discovery.Helper,
-	resticBackupper restic.Backupper,
-	resticSnapshotTracker *pvcSnapshotTracker,
-	volumeSnapshotterGetter VolumeSnapshotterGetter,
-) ItemBackupper {
-	ib := &defaultItemBackupper{
-		backupRequest:           backupRequest,
-		tarWriter:               tarWriter,
-		dynamicFactory:          dynamicFactory,
-		discoveryHelper:         discoveryHelper,
-		resticBackupper:         resticBackupper,
-		resticSnapshotTracker:   resticSnapshotTracker,
-		volumeSnapshotterGetter: volumeSnapshotterGetter,
-
-		itemHookHandler: &defaultItemHookHandler{
-			podCommandExecutor: podCommandExecutor,
-		},
-	}
-
-	// this is for testing purposes
-	ib.additionalItemBackupper = ib
-
-	return ib
-}
-
+// ItemBackupper can back up individual items to a tar writer.
 type ItemBackupper interface {
```

Comment:
TODO this interface probably isn't needed anymore since I've removed the factory, and am going to remove the additionalItemBackupper field.